browser/ui/views/page_info/brave_shields_page_info_view.cc

```diff
@@ -0,0 +1,152 @@
+#include "content/public/browser/navigation_controller.h"
+#include "content/public/browser/web_contents.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/gfx/geometry/rounded_corners_f.h"
+#include "ui/views/controls/webview/unhandled_keyboard_event_handler.h"
+#include "ui/views/controls/webview/webview.h"
+#include "ui/views/layout/fill_layout.h"
+#include "url/gurl.h"
+
+namespace {
+
+class ShieldsWebView : public views::WebView,
+                       public WebUIContentsWrapper::Host {
+  METADATA_HEADER(ShieldsWebView, views::WebView)
+
+ public:
+  ShieldsWebView(base::WeakPtr<WebUIContentsWrapper> contents_wrapper,
+                 base::RepeatingCallback<void()> close_bubble)
+      : contents_wrapper_(std::move(contents_wrapper)),
+        close_bubble_(std::move(close_bubble)) {
+    contents_wrapper_->SetHost(weak_factory_.GetWeakPtr());
+    SetWebContents(contents_wrapper_->web_contents());
+    holder()->SetCornerRadii(gfx::RoundedCornersF(0, 0, 16, 16));
+  }
+
+  ~ShieldsWebView() override = default;
+
+  // WebUIContentsWrapper::Host:
+  void ShowUI() override {}
+  void CloseUI() override { close_bubble_.Run(); }
+
+  // views::WebView:
+  bool HandleContextMenu(content::RenderFrameHost& render_frame_host,
+                         const content::ContextMenuParams& params) override {
+    return true;
+  }
+
+  void VisibilityChanged(views::View* starting_from, bool is_visible) override {
+    views::WebView::VisibilityChanged(starting_from, is_visible);
+    if (contents_wrapper_) {
+      if (auto* web_contents = contents_wrapper_->web_contents()) {
+        if (is_visible) {
+          web_contents->WasShown();
+        } else {
+          web_contents->WasHidden();
+        }
+      }
+    }
+  }
+
+  bool HandleKeyboardEvent(
+      content::WebContents* source,
+      const input::NativeWebKeyboardEvent& event) override {
+    return unhandled_keyboard_event_handler_.HandleKeyboardEvent(
+        event, GetFocusManager());
+  }
+
+  void ResizeDueToAutoResize(content::WebContents* source,
+                             const gfx::Size& new_size) override {
+    SetPreferredSize(new_size);
+  }
+
+ private:
+  base::WeakPtr<WebUIContentsWrapper> contents_wrapper_ = nullptr;
+  base::RepeatingCallback<void()> close_bubble_;
+  views::UnhandledKeyboardEventHandler unhandled_keyboard_event_handler_;
+  base::WeakPtrFactory<ShieldsWebView> weak_factory_{this};
+};
+
+BEGIN_METADATA(ShieldsWebView)
+END_METADATA
+
+void LoadShieldsURL(content::NavigationController& nav_controller,
+                    std::string_view query) {
+  nav_controller.LoadURL(GURL(kShieldsPanelURL + std::string(query)),
+                         content::Referrer(), ui::PAGE_TRANSITION_AUTO_TOPLEVEL,
+                         /*extra_headers=*/std::string());
+}
+
+}  // namespace
+
+BraveShieldsPageInfoView::BraveShieldsPageInfoView(
+    BrowserWindowInterface* browser_window_interface,
+    content::WebContents* web_contents,
```

Comment:
I think we don't need this |web_contents| arg if it's only for fetching BrowserContext. `BrowserWindowInterface::GetProfile()` can give it.
Or any plan for using it in f/u?